lib/bolt/pal.rb

```diff
@@ -143,6 +143,33 @@ def list_tasks
       end
     end
 
+    def parse_param(str, type, compiler)
+      begin
+        parsed = JSON.parse(str, quirks_mode: true)
+        type.instance?(parsed) ? parsed : str
```

Comment:
This seems reasonable, and like something we can reproduce with pcore-java.

Reply:
We can reproduce it with pcore-java but that means that anything in puppet-task or the console  needs to be parsed server side.

Reply:
In general this approach seems robust.